### PR TITLE
Stream supervision doc clarification

### LINKED ISSUE #23376: Document what stages implement supervision
Currently there isn't really a way to know what stages care about the supervision strategy attribute. 

We should document clearly that it isn't omnipresent like it is with actors here, and also update the "these are the three stages that doesn't support supervision yet": http://doc.akka.io/docs/akka/current/scala/stream/stream-error.html#error-handling

We should also add in the scaladoc on for each stage if it does heed the supervision strategy or not.

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #23376

### CODE REVIEW

File: akka-docs/src/test/java/jdocs/stream/FlowErrorDocTest.java [hunk 1]
```diff
@@ -138,4 +139,37 @@ public void demonstrateRestartSectionStream() throws Exception {
...
+    final Materializer mat = ActorMaterializer.create(system);
+    Source.from(Arrays.asList(0, 1, 2, 3, 4, 5, 6)).map(n -> {
+      if (n < 5) return n.toString();
+      else throw new RuntimeException("Boom!");
+    }).recover(new PFBuilder()
+        .match(RuntimeException.class, ex -> "stream truncated")
+        .build()
+    ).runForeach(System.out::println, mat);
+    //#recover
+  }
```
Comment (ktoso) [inline akka-docs/src/test/java/jdocs/stream/FlowErrorDocTest.java]: good examples

File: akka-stream/src/main/scala/akka/stream/Attributes.scala [hunk 2]
```diff
@@ -297,12 +297,18 @@ object ActorAttributes {
 
   /**
    * Scala API: Decides how exceptions from user are to be handled.
+   *
+   * Note that this is implemented on a per stage basis. Stages supporting supervision has
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/Attributes.scala]: "have"

though perhaps more explicitly

`Stages supporting supervision strategies explicitly document that they do so. If a stage does not document support for these, it should be assume to not support supervision.`

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 3]
```diff
@@ -406,6 +406,8 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * Transform this stream by applying the given function to each of the elements
    * as they pass through this processing step.
    *
+   * Adheres to the supervision strategy attribute.
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Yeah good.. perhaps link to it? `[[]]`?

**Review by ktoso — APPROVED** [review]
LGTM, lookin good

**Comment by ktoso** [comment]
(just some minor comments)

File: akka-docs/src/main/paradox/scala/stream/stream-error.md [hunk 4]
```diff
@@ -1,18 +1,99 @@
...
+Recovering can be useful if you want to gracefully complete a stream on failure while letting 
+downstream know that there was a failure.
+
+Scala
+:   @@snip [FlowErrorDocSpec.scala]($code$/scala/docs/stream/FlowErrorDocSpec.scala) { #recover }
+
+Java
+:   @@snip [FlowErrorDocTest.java]($code$/java/jdocs/stream/FlowErrorDocTest.java) { #recover }
+
+This will output:
```
Comment (johanandren) [inline akka-docs/src/main/paradox/scala/stream/stream-error.md]: Good idea

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala (hunk 3 above)
Comment (johanandren) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Good idea.

**Comment by johanandren** [comment]
Waiting for the restarting stages now.

**Comment by ktoso** [comment]
You can try to take over, or merge and fix formatting right away - the only issue there was the formatting failure i think

File: akka-docs/src/main/paradox/scala/stream/stream-error.md [hunk 5]
```diff
@@ -1,18 +1,134 @@
...
+Scala
+:   @@snip [FlowErrorDocSpec.scala]($code$/scala/docs/stream/FlowErrorDocSpec.scala) { #recoverWithRetries-output }
+
+Java
+:   @@snip [FlowErrorDocTest.java]($code$/java/jdocs/stream/FlowErrorDocTest.java) { #recoverWithRetries-output }
+
+
+## Delayed restarts with a backoff stage
+
+Just as Akka provides the @ref:[backoff supervision pattern for actors](../general/supervision.md#backoff-supervisor), Akka streams
```
Comment (raboof) [inline akka-docs/src/main/paradox/scala/stream/stream-error.md]: Do we prefer `@ref[` over `@ref:[`?

**Review by raboof — APPROVED** [review]

File: akka-docs/src/main/paradox/scala/stream/stream-error.md (hunk 5 above)
Comment (johanandren) [inline akka-docs/src/main/paradox/scala/stream/stream-error.md]: I have no opinion I just copy-pasta

Comment (johanandren) [inline akka-docs/src/main/paradox/scala/stream/stream-error.md]: Does it mean different things or is it alternativ syntax for the same feature?

Comment (raboof) [inline akka-docs/src/main/paradox/scala/stream/stream-error.md]: IIRC it's different ways to write the same - http://developer.lightbend.com/docs/paradox/latest/features/linking.html suggests `@ref:[` though - perhaps `@ref[` will not be recognized as a link by github? If so we should prefer `@ref:[` ;).

**Pull request merged**